src/main/java/com/unstoppabledomains/resolution/contracts/HTTPUtil.java

```diff
@@ -0,0 +1,64 @@
+package com.unstoppabledomains.resolution.contracts;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+public class HTTPUtil {
+    protected JsonObject post(String url, String address, String data) throws IOException {
```

Comment:
It is better approached using `static` methods instead of inheritence. See, java doesn't support multiple inheritence so it is better to be kept for cases where you really need inheritence features like `abstract` or `super`.